### PR TITLE
Migrating namedstruct tests from bgtfs to namedstruct + py2-3 support

### PR DESCRIPTION
**Pull request by annaelleduff**

### CODE REVIEW

File: src/bits.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "bits.h"
```
Comment (gcamp) [inline src/bits.cpp]: Could be nice to have a little comment on why we do this.

File: values.py [hunk 2]
```diff
@@ -790,7 +812,7 @@ def pretty(self):
     
     def getImmediateDataSize(self):
         fieldLengths = self.getFieldLengths()
-        return (len(fieldLengths) + 2) * 2 + (sum(fieldLengths) * len(self.entries) + 7) / 8
+        return (len(fieldLengths) + 2) * 2 + old_div((sum(fieldLengths) * len(self.entries) + 7), 8)
```
Comment (gcamp) [inline values.py]: What is old_div?

File: src/namedStructTests.h [hunk 3]
```diff
@@ -0,0 +1,1968 @@
+/* Testing structs generated by namedstructpy */
```
Comment (gcamp) [inline src/namedStructTests.h]: Actually just realized that this file is created automatically. Maybe not commit it and assume it's created automatically? Make sure it's also created automatically at the right place?

File: values.py (hunk 2 above)
Comment (annaelleduff) [reply values.py]: https://github.com/TransitApp/namedstruct/pull/1/files/f9532c720726e1a56d4972e53783d49c625c5922#diff-6727fca2a6f06a2a33201d4300d43f0cR7

old division, it's been added by the script to migrate from py2 to py3

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        namedStructTests.mm
+        testBits.mm
+    )
+
+    set_property(TARGET namedstructTests PROPERTY CXX_STANDARD 14)
+    set_property(TARGET namedstructTests PROPERTY CXX_STANDARD_REQUIRED ON)
+
+    set_source_files_properties(${PROJECT_SOURCE_DIR}/localTestFiles PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
+
+    # add_custom_command(
```
Comment (gcamp) [inline src/CMakeLists.txt]: You can replace this command by the create test files script

File: types.py [hunk 5]
```diff
@@ -591,7 +594,12 @@ def __init__(self, name, enumType, mapping):
         for name, value in mapping:
             stringhelper.assertIsValidIdentifier(name)
             namedstructValue = enumType.makeValue(value)
-            self._hasNegativeValues = self._hasNegativeValues or (value < 0)
+            # for py3 compatibility
```
Comment (gcamp) [inline types.py]: is everything in there all python 2 compatible? If not, we can drop the compatibility check right?

Comment (annaelleduff) [reply types.py]: I can delete the comment but I wouldn't take the risk of deleting the if / else

**Pull request merged**